molecule/default/tasks/openshift_adm_prune_builds.yml

```diff
@@ -0,0 +1,153 @@
+- block:
+  - set_fact:
+      build_ns: "prune-builds"
+      build_config_name: "ruby-centos-ansible"
+      is_name: "ruby-centos"
+
+  - name: Ensure namespace
+    community.okd.k8s:
+      kind: Namespace
+      name: "{{ build_ns }}"
+
+  - name: Create image stream
+    community.okd.k8s:
+      namespace: "{{ build_ns }}"
+      definition:
+        apiVersion: image.openshift.io/v1
+        kind: ImageStream
+        metadata:
+          name: "{{ is_name }}"
+        spec:
+          lookupPolicy:
+            local: false
+          tags: []
+
+  - name: Create build configuration
+    community.okd.k8s:
+      namespace: "{{ build_ns }}"
+      definition:
+        kind: BuildConfig
+        apiVersion: build.openshift.io/v1
+        metadata:
+          name: "{{ build_config_name }}" 
+        spec:
+          source: 
+            dockerfile: |
+              FROM openshiftfake/ruby-fake-centos7
+              USER ansible
+          strategy:
+            type: Docker
+          output: 
+            to:
+              kind: "ImageStreamTag"
+              name: "{{ is_name }}:latest"
+
+  - name: Start Build from Build config waiting for build to complete
+    community.okd.openshift_start_build:
+      build_config_name: "{{ build_config_name }}"
+      namespace: "{{ build_ns }}"
+      wait: yes
+      wait_timeout: 120
+    register: new_build
+    ignore_errors: yes
+
+  - assert:
+      that:
+        - new_build is failed
+
+  - name: Retrieve Build information
+    kubernetes.core.k8s_info:
+      version: "build.openshift.io/v1"
+      kind: Build
+      namespace: "{{ build_ns }}"
+      name: "{{ build_config_name }}-1"
+    register: new_build
+
+  - name: Assert Build is neither pending nor running
+    assert:
+      that:
+        - new_build.resources.0.status.phase == 'Failed'
+
+  - name: Prune Builds keep younger than 30min
```

Comment:
Could you also please add idempotency checks for the tests (where it's sensible to have)?

Reply:
will do thanks